# HG changeset patch
# User rk
# Date 1285857046 -7200
# Node ID 9a535123d884b23ebe38968c6b346f3575d615cb
# Parent  c9985319230dcb037cd96a3b5306ef9f0ae3bd69
when calling wxAuiManager::LoadPerspective() all panes got docked even those that were created to be non-dockable

diff -r c9985319230d -r 9a535123d884 src/aui/framemanager.cpp
--- a/src/aui/framemanager.cpp	Thu Sep 30 15:34:06 2010 +0200
+++ b/src/aui/framemanager.cpp	Thu Sep 30 16:30:46 2010 +0200
@@ -1570,10 +1570,15 @@ bool wxAuiManager::LoadPerspective(const
     if (part != wxT("layout2"))
         return false;
 
-    // mark all panes currently managed as docked and hidden
+    // Mark all panes currently managed as hidden. Also, dock all panes that are dockable.
     int pane_i, pane_count = m_panes.GetCount();
     for (pane_i = 0; pane_i < pane_count; ++pane_i)
-        m_panes.Item(pane_i).Dock().Hide();
+    {
+        wxAuiPaneInfo& p = m_panes.Item(pane_i);
+        if(p.IsDockable())
+            p.Dock();
+        p.Hide();
+    }
 
     // clear out the dock array; this will be reconstructed
     m_docks.Clear();
